data/store/mongo/mongo_datum.go

```diff
@@ -739,18 +711,23 @@ func (d *DatumRepository) populateLastUpload(ctx context.Context, userId string,
 	return nil
 }
 
-func (d *DatumRepository) populateEarliestModified(ctx context.Context, userId string, typ string, status *types.UserLastUpdated) (err error) {
+func (d *DatumRepository) populateEarliestModified(ctx context.Context, userId string, typ []string, status *data.UserDataStatus) (err error) {
 	// get earliest modified record which is newer than LastUpdated
 	selector := bson.M{
-		"_active": bson.M{"$ne": -1111},
 		"_userId": userId,
-		"type":    typ,
+		"_active": bson.M{"$in": bson.A{true, false}},
 		"time": bson.M{
 			"$gte": status.FirstData,
 			"$lte": status.LastData,
 		},
 	}
 
+	if len(typ) > 1 {
```

Comment:
The special case here should be `len(typ) == 1`, because `$in` can handle empty arrays. Currently, there is a bug when `typ` is `nil` or `empty`. In addition to adding a guard for the length I think you should flip the logic.

Reply:
done